tracer/src/Datadog.Trace/AppSec/Security.cs

```diff
@@ -24,15 +27,54 @@ internal class Security : IDatadogSecurity, IDisposable
     {
         private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor<Security>();
 
+        private static readonly Dictionary<string, string> RequestHeaders;
+        private static readonly Dictionary<string, string> ResponseHeaders;
+
         private static Security _instance;
         private static bool _globalInstanceInitialized;
         private static object _globalInstanceLock = new();
 
         private readonly IWaf _waf;
-        private readonly IAppSecAgentWriter _agentWriter;
         private readonly InstrumentationGateway _instrumentationGateway;
         private readonly SecuritySettings _settings;
 
+        static Security()
+        {
+            var requestHeaders = new List<string>()
```

Comment:
[nit] Why don't you populate the dictionary right away instead of going through a list? Even if it's a static initialize it's still less allocations and I don't find it more cumbersome or less readable to have something like:
```
RequestHeader = new Dictionary<string, string>(18)
  {
      {"X-FORWARDED-FOR", string.Empty},
      {"X-CLIENT-IP", string.Empty},
      {"X-REAL-IP", string.Empty},
      {"X-FORWARDED", string.Empty},
      {"X-CLUSTER-CLIENT-IP", string.Empty},
      {"FORWARDED-FOR", string.Empty},
      {"FORWARDED", string.Empty},
      {"VIA", string.Empty},
      {"TRUE-CLIENT-IP", string.Empty},
      {"Content-Length", string.Empty},
      {"Content-Type", string.Empty},
      {"Content-Encoding", string.Empty},
      {"Content-Language", string.Empty},
      {"Host", string.Empty},
      {"user-agent", string.Empty},
      {"Accept", string.Empty},
      {"Accept-Encoding", string.Empty},
      {"Accept-Language", string.Empty},
  };
```